src/CMakeLists.txt

```diff
@@ -15,5 +15,20 @@
 # specific language governing permissions and limitations
 # under the License.
 
+add_subdirectory(arrow)
 add_subdirectory(core)
 add_subdirectory(puffin)
+
+include(CMakePackageConfigHelpers)
+
+configure_package_config_file("${CMAKE_CURRENT_SOURCE_DIR}/config.cmake.in"
+                              "${CMAKE_CURRENT_BINARY_DIR}/iceberg-config.cmake"
+                              INSTALL_DESTINATION "${ICEBERG_INSTALL_CMAKEDIR}/Iceberg")
+
+write_basic_package_version_file(
+  "${CMAKE_CURRENT_BINARY_DIR}/iceberg-config-version.cmake"
+  COMPATIBILITY SameMajorVersion)
+
+install(FILES "${CMAKE_CURRENT_BINARY_DIR}/iceberg-config.cmake"
```

Comment:
FYI, the install directory (with only static library built) looks like below:
```
/tmp/iceberg/
├── lib/
│   ├── libarrow.a
│   ├── libiceberg_arrow.a
│   ├── libiceberg_core.a
│   ├── libiceberg_puffin.a
│   └── cmake/
│       ├── Iceberg/
│       │   ├── iceberg-arrow-targets.cmake
│       │   ├── iceberg-arrow-targets-debug.cmake
│       │   ├── iceberg-core-targets.cmake
│       │   ├── iceberg-core-targets-debug.cmake
│       │   ├── iceberg-puffin-targets.cmake
│       │   ├── iceberg-puffin-targets-debug.cmake
│       │   ├── iceberg-config.cmake
│       │   └── iceberg-config-version.cmake
├── include/
│   └── iceberg/
│       ├── puffin.h
│       ├── table.h
│       └── demo_arrow.h
└── share/
    └── doc/
        └── iceberg/
            ├── LICENSE
            └── NOTICE
```

Reply:
It seems the demo header shouldn't be installed?

Reply:
The demo_arrow.h is included by the example which consumes the iceberg library via installed location. I add `demo_` prefix by purpose to show this and it will be removed once we have more source files.

Reply:
Why do we need to install arrow static lib? How do we suppose a user to use it while arrow headers is not installed? dynamic lib might be useful at linking stage, what the static lib is for? I might be too dumb to follow you 😂

Reply:
We need to install the arrow static lib only when we provided vendored arrow in the build process. There are several cases:

1. If users run `cmake .. -DICEBERG_ARROW=OFF`, then we get rid of Arrow at all.
2. If users run `cmake .. -DICEBERG_ARROW=ON`, then we vendor and install static Arrow lib.
3. If users have installed arrow already, then they simply run `cmake .. -DCMAKE_PREFIX_PATH=/path/to/arrow -DICEBERG_ARROW=ON ` to point to that location and `FetchContent_MakeAvaiable` is smart enough to locate it. In this case, we don't install Arrow lib and it is the preferred use case in production.

BTW, I have added components support in this PR as well. If users want to make sure iceberg_arrow is available, they can call `find_package(Iceberg REQUIRED CONFIG COMPONENTS Core Arrow)`.